### PR TITLE
Support fp16 in estimator NCF model

### PR DESCRIPTION
**Pull request by reedwm**
fp16 is added with the new `tf.train.experimental.enable_mixed_precision_graph_rewrite` function.

Note: I moved the `get_loss_scale` function from _performance.py to resnet_run_loop.py, and added a similar function to the NCF model. This is because the default loss scale should be different for each model.

To test convergence, I ran:

```
python ncf_estimator_main.py --model_dir ~/ncf_model_dir --data_dir ~/ncf_data_dir --dataset=ml-20m --hooks= --num_gpus=1 --clean --train_epochs=14 --batch_size=98340 --learning_rate=0.00382059 --beta1=0.783529 --beta2=0.909003 --epsilon=1.45439e-07 --layers=256,256,128,64 --num_factors=64 --hr_threshold=0.635 --ml_perf --dtype=fp16
```
I got an HR of 0.6371 on epoch 8.

Performance gain is modest: From about 67 steps/sec to 75 steps/sec with a batch size of 98340 on a V100.

### CODE REVIEW

**Comment by guptapriya** [comment]
Hey @reedwm sorry i missed this PR. Should I review it now or you plan to update it?

**Comment by reedwm** [comment]
I'll update it once I get time. Closing for now until I update it.
